cpp/CMakeLists.txt

```diff
@@ -515,6 +515,7 @@ endif()
 set(ARROW_BENCHMARK_LINK_LIBS
   arrow_static
   arrow_benchmark_main
+  gtest
```

Comment:
Was this ... the missing link?

Reply:
In all seriousness, was this actually missing before this PR? Or is it now necessary because of changes made in this PR?

Reply:
It's necessary bc of changes in this PR, uses symbols that aren't header only from gtest

Reply:
Cool. Thanks for clarifying.